httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/PoolingHttpClientConnectionManager.java

```diff
@@ -323,7 +323,7 @@ public synchronized ConnectionEndpoint get(
                         final TimeValue timeToLive = connectionConfig.getTimeToLive();
                         if (TimeValue.isNonNegative(timeToLive)) {
                             final Deadline deadline = Deadline.calculate(poolEntry.getCreated(), timeToLive);
-                            if (deadline.isExpired()) {
+                            if (timeValue.getDuration() == 0 || deadline.isExpired()) {
```

Comment:
@ChristophKaser 
1. What is the point of calculating the deadline if it is not going be used? This looks wrong. Please refactor the code to avoid unnecessary deadline calculation. There are various ways of accomplishing that
2. Please make symmetric changes to the `PoolingAsyncClientConnectionManager`
3. Please make sure that the build does not fail

Reply:
1. Done - I have inlined the deadline calculation now.
2. Done
3. Ouch, very sorry about that. I had made the changes to an internal fork and copied them into the vanilla project, but did not run `mvn verify` after that. I have corrected the typo (and this time, I did run mvn).